raiden/transfer/mediated_transfer/initiator.py

```diff
@@ -398,21 +398,30 @@ def handle_onchain_secretreveal(
     the current lock removed from the merkle tree and the transferred amount
     updated.
     """
+    secret = state_change.secret
+    secrethash = initiator_state.transfer_description.secrethash
     is_valid_secret = is_valid_secret_reveal(
         state_change=state_change,
-        transfer_secrethash=initiator_state.transfer_description.secrethash,
-        secret=state_change.secret,
+        transfer_secrethash=secrethash,
+        secret=secret,
     )
     is_channel_open = channel.get_status(channel_state) == CHANNEL_STATE_OPENED
     is_lock_expired = state_change.block_number > initiator_state.transfer.lock.expiration
 
     is_lock_unlocked = (
         is_valid_secret and
-        is_channel_open and
         not is_lock_expired
     )
 
     if is_lock_unlocked:
+        channel.register_onchain_secret(
```

Comment:
I don't understand under which circumstances the `initiator` would _want_ to register on-chain?
- for a sent transfer, it will lose the sent locked tokens it otherwise might get back plus gas
- for a refund transfer, it would be zero sum (sent locked tokens - refund locked tokens == 0) plus gas

Reply:
This is just registering the fact that the secret was registered on chain, but it is not requesting it to happen.